dali/image/jpeg.cc

```diff
@@ -25,35 +26,35 @@ JpegImage::JpegImage(const uint8_t *encoded_buffer,
                      DALIImageType image_type)
   : GenericImage(encoded_buffer, length, image_type) {
 }
-#if !defined(DALI_USE_JPEG_TURBO)
-bool get_jpeg_size(const uint8 *data, size_t data_size, int *height, int *width) {
+
+#ifndef DALI_USE_JPEG_TURBO
+bool get_jpeg_size(const uint8 *data, size_t data_size, int *height, int *width, int *nchannels) {
   // Check for valid JPEG image
-  unsigned int i = 0;  // Keeps track of the position within the file
-  if (data[i] == 0xFF && data[i + 1] == 0xD8) {
-    i += 4;
-    // Retrieve the block length of the first block since
-    // the first block will not contain the size of file
-    uint16_t block_length = data[i] * 256 + data[i + 1];
-    while (i < data_size) {
-      i += block_length;  // Increase the file index to get to the next block
-      if (i >= data_size) return false;  // Check to protect against segmentation faults
-      if (data[i] != 0xFF) return false;  // Check that we are truly at the start of another block
-      if (data[i + 1] >= 0xC0 && data[i + 1] <= 0xC3) {
-        // 0xFFC0 is the "Start of frame" marker which contains the file size
-        // The structure of the 0xFFC0 block is quite simple
-        // [0xFFC0][ushort length][uchar precision][ushort x][ushort y]
-        *height = data[i + 5] * 256 + data[i + 6];
-        *width = data[i + 7] * 256 + data[i + 8];
-        return true;
-      } else {
-        i += 2;  // Skip the block marker
-        block_length = data[i] * 256 + data[i + 1];  // Go to the next block
-      }
-    }
-    return false;  // If this point is reached then no size was found
-  } else {
+  if (ReadValueBE<uint16_t>(data) != 0xFFD8)
```

Comment:
Checking two bytes was in line with the wikipedia description of SOI marker as bytes 0xFF, 0xF8 (not a 16-bit number 0xFFD8)

Reply:
:+1: